rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
+  if (!publisher->topic_name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory for node name");
+    goto fail;
+  }
+  memcpy(const_cast<char *>(publisher->topic_name), topic_name, strlen(topic_name) + 1);
```

Comment:
I realized why we don't use strncpy everywhere, it's because it provokes a warning on Windows. Yarr. http://ci.ros2.org/job/ci_windows/1110/warnings34Result/new/package.-203766663/source.6589365479437991052/#1133